effectful/handlers/llm/providers.py

```diff
@@ -55,6 +55,32 @@ def _pil_image_to_openai_image_param(
     }
 
 
+@defop
+@functools.singledispatch
+def format_value(value: Any) -> OpenAIMessageContent:
+    """Convert a Python value to internal message part representation.
+
+    This function can be extended by registering handlers for
+    different types using @format_value.register.
+
+    Returns a OpenAIMessageContent - either a string or a list of OpenAIMessageContentListBlock.
+    """
+    return [{"type": "text", "text": str(value)}]
+
+
+@format_value.register(Image.Image)  # type: ignore
+def _(value: Image.Image) -> OpenAIMessageContent:
+    return [_pil_image_to_openai_image_param(value)]
+
+
+@format_value.register(Sequence)  # type: ignore
+def _(values: Sequence) -> OpenAIMessageContent:
+    if all(isinstance(value, Image.Image) for value in values):
+        return [_pil_image_to_openai_image_param(value) for value in values]
+    else:
+        return [{"type": "text", "text": str(values)}]
```

Comment:
This will capture `str`, which isn't what we want.

Reply:
What do you mean by capturing str? and what encoding would you suggest instead?

Reply:
done!